### DISCUSSION TITLE
Concourse web is running but fly command is not connecting

### DISCUSSION THREAD

**Discussion by abhipsnl** [concourse/concourse#5552] [General]
I am not sure what is wrong here, but my concourse web service is running.
and when I do fly -t main workers , it giving me 
could not reach the Concourse server called quolls:

    Get https://cicdxxxx/api/v1/info: dial tcp 10.xxxx:443: connect: connection refused

is the targeted Concourse running? better go catch it lol

So once I restart my web , it start working , but this is very frequent , how to troubleshoot , not much help from logs

**Comment by vito** [comment]
Hmm sorry but this looks like a pretty primitive network failure that would be pretty hard for anyone to diagnose and troubleshoot from afar. "connection refused" means, well, `fly` couldn't connect. This could be because the `web` node isn't running - maybe it crashed?

**Reply by abhipsnl** [reply]
Actually I thought alot before writing this  
Service is running,  but fly don't connect,  just after restarting the web(master node) it strat working   I know its tough to debug with such statements but is it possible to get detailed logs why these happened

**Comment by jamieklassen** [comment]
you can set the log level to debug, if you haven't already done that.
* does the web UI work? how did you determine the "service is running"?
* how is your concourse deployed (k8s/bosh/docker/systemd/etc), and what are all the flags you are using?
* it looks like fly is connecting over https - is concourse handling the TLS termination, or maybe you're using a load balancer/reverse proxy?
* when you say it "works after a restart" what does that mean? that `fly workers` correctly displays the workers with no error?
* you also say the error happens frequently. does this mean that you restart, it starts working, and some indefinite time later it stops working so you restart?

**Comment by skreddy6673** [comment]
Most times, when i see this error is due to network issues between the system from which i'm running the fly login command and the concourse server. I would say closing the terminal from which you are seeing the error and trying in a new terminal will help.

If you see the web node running and you can access concourse at https://cicdxxxx. I would say you don't need to restart the web node. Just close the terminal and try it from a new terminal.
